docs/bir-spec/src/main/resources/kaitai/bir.ksy

```diff
@@ -176,18 +176,41 @@ types:
       - id: value
         size: value_length
   type_invokable:
+    seq:
+      - id: invokable_kind
+        type: u1
+        enum: invokable_kind_enum
+      - id: cp_info
+        type:
+          switch-on: invokable_kind
+          cases:
+            'invokable_kind_enum::invokable_kind_any': any_function
+            'invokable_kind_enum::invokable_kind_specific': type_invokable_body
+    enums:
+      invokable_kind_enum:
+        0: invokable_kind_specific
+        1: invokable_kind_any
+  any_function:
+    seq:
+      - id: any_function_type
+        type: u1
```

Comment:
Can't we do the following here and get rid of writing the `isAnyFunction` boolean to the BIR twice? https://github.com/ballerina-platform/ballerina-lang/pull/28233/files#r591448728

```suggestion
    seq: []
```

https://doc.kaitai.io/user_guide.html#do-nothing

Reply:
@manuranga, @Kishanthan, @warunalakshitha, can you please let us know the correct approach here? 

I don't think we should be duplicating `isAnyFunction` in the BIR at https://github.com/ballerina-platform/ballerina-lang/blob/49b99ab21ce99f8b53a516ec1ea0e2b540f3901e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java#L189 just for this.

Reply:
Yeah, I think we don't need to write twice. Since you already write the flag (0 or 1), you can read it, and based on the value, read the rest of binary (in here it is the section read the rest param info), else skip and go to the next element to read. 

You can check this example - https://github.com/ballerina-platform/ballerina-lang/blob/master/docs/bir-spec/src/main/resources/kaitai/bir.ksy#L205, where we read the flag and then read the body if the value is 1.